### PR TITLE
8255672: Replace PhaseTransform::eqv by pointer equality check

### PR DESCRIPTION
**Pull request by TobiHartmann**
`PhaseTransform::eqv(n1, n2)` can be replaced by `n1 == n2`. Code in adlc also refers to it but already uses `==` instead.

Thanks,
Tobias

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8255672](https://bugs.openjdk.java.net/browse/JDK-8255672): Replace PhaseTransform::eqv by pointer equality check

### Reviewers
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/999/head:pull/999`
`$ git checkout pull/999`

### CODE REVIEW

File: src/hotspot/share/opto/callnode.cpp [hunk 1]
```diff
@@ -1224,8 +1224,12 @@ Node* SafePointNode::Identity(PhaseGVN* phase) {
...
 //------------------------------Value------------------------------------------
 const Type* SafePointNode::Value(PhaseGVN* phase) const {
-  if( phase->type(in(0)) == Type::TOP ) return Type::TOP;
-  if( phase->eqv( in(0), this ) ) return Type::TOP; // Dead infinite loop
+  if (phase->type(in(0)) == Type::TOP) {
+    return Type::TOP;
+  }
+  if (in(0) == this) {
+    return Type::TOP; // Dead infinite loop
+  }
```
Comment (chhagedorn) [inline src/hotspot/share/opto/callnode.cpp:1232]: Could be merged with the `if` above but might be clearer if left as it is now.

**Review by chhagedorn — APPROVED** [review]
Looks good to me!

File: src/hotspot/share/opto/subnode.cpp [hunk 2]
```diff
@@ -445,11 +451,10 @@ Node *SubFNode::Ideal(PhaseGVN *phase, bool can_reshape) {
...
   // Not associative because of boundary conditions (infinity)
-  if( IdealizedNumerics && !phase->C->method()->is_strict() ) {
+  if (IdealizedNumerics && !phase->C->method()->is_strict() &&
+      in(2)->is_Add() && in(1) == in(2)->in(1)) {
     // Convert "x - (x+y)" into "-y"
-    if( in(2)->is_Add() &&
-        phase->eqv(in(1),in(2)->in(1) ) )
-      return new SubFNode( phase->makecon(TypeF::ZERO),in(2)->in(2));
+    return new SubFNode(phase->makecon(TypeF::ZERO),in(2)->in(2));
   }
```
[the remark this answers is not in the document]
Comment (TobiHartmann) [reply src/hotspot/share/opto/subnode.cpp:458]: Good catch!

**Comment by TobiHartmann** [comment]
Thanks for the review Christian. I'd like to not merge the ifs because I think these are separate cases and I would like to keep the impact on surrounding code to a minimum. Are you okay with that?

**Review by chhagedorn — APPROVED** [review]
Yes, I think that is reasonable to leave the code as it is and only do the clean-ups directly related to `PhaseTransform::eqv`. Looks good!

**Review by cl4es — APPROVED** [review]
Nice cleanup!

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by TobiHartmann** [comment]
@chhagedorn, @cl4es, @vnkozlov thanks for the reviews!

/integrate

**Pull request closed without merging**